### PR TITLE
Add Spectacle documentation

### PR DESCRIPTION
**Pull request by smizell**
This PR focuses on Spectacle documentation and making GraphiQL more accessible to the users.

* Adds a CLI command `api spectacle` for opening up GraphiQL in the browser
* Adds documentation to the GraphQL schema
* Adds a page in the Optic documentation
* Adds a note in the Scripting documentation page about the `SPECTACLE_URL` environment variable

Note: there are a few places in the schema that I wasn't for sure about the description, so I've left them out for now.

### CODE REVIEW

File: website/docs/using/spectacle.mdx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    body: JSON.stringify(body),
+    headers: { "Content-Type": "application/json" },
+  });
+  return await result.json();
+}
+```
+
+1. Install [Node.js](https://nodejs.org/en/) if you haven't already
+1. Save this code above as `spectacle.js`
+1. Install `node-fetch` with `npm install node-fetch`
```
Comment (devdoshi) [inline website/docs/using/spectacle.mdx:67]: i've never done this without doing npm init first, does it work?

**Review by devdoshi — APPROVED** [review]
Looks good to me. If we need something richer, we can expose graphiql on the frontend as well, so we can use it in our browser demos. https://github.com/graphql/graphiql/blob/main/packages/graphiql/README.md#usage-npm-module

File: website/docs/using/spectacle.mdx (hunk 1 above)
Comment (smizell) [reply website/docs/using/spectacle.mdx:67]: Yeah, it creates the `package.json` with just the dependencies:

```js
{
  "dependencies": {
    "node-fetch": "^2.6.1"
  }
}
```

**Comment by smizell** [comment]
Oh, I like that idea, too. That would make it even more visible.

File: website/docs/using/spectacle.mdx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+We'll look at an example using a Node.js script. The one below is using the `requests` query to get information about requests and responses in the spec. It's using `node-fetch` as the library to query Spectacle.
+
+```javascript
+const fetch = require("node-fetch");
+
+main();
+
+async function main() {
+  const result = await querySpectacle({
```
Comment (acunniffe) [inline website/docs/using/spectacle.mdx:31]: @smizell -- shouldn't we show people how to get the `querySpectacle` to work? Do they need to import the Spectacle package? something else? 

Might be good to expose a function in the package that looks for the ENV in script and just makes it dead simple to call it.l

**Pull request merged**